File: Two Pointer/Squares_of_a_sorted_Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& A) {
       
        int len =A.size();
        int left = 0, right = len-1;
        vector<int> res(len);
        for(int i = len-1;i>=0;i--)
        {
            int left_sqr = A[left]*A[left];
            int right_sqr =  A[right]*A[right];
            if(left_sqr >= right_sqr)
            {
                res[i] = left_sqr;
                left++;
            }
            else
            {
                res[i] = right_sqr;
                right--;
            }
            
        }
        return res;
    }
};

File: Two Pointer/Java/Easy/Squares_of_a_SortedArray.java
/* -------- O(N) ------ */

class Solution {
    public int[] sortedSquares(int[] nums) {
        
        int[] sortedArr = new int[nums.length];
        int left =0, right = nums.length -1 , i= nums.length-1;
        while(left<= right)
        {
            if(Math.abs(nums[left]) > Math.abs(nums[right]))
            {
                sortedArr[i] = nums[left] * nums[left];
                left++;
            }
            else
            {
                sortedArr[i] = nums[right] * nums[right];
                right--;
            }
            i--;
        }
        return sortedArr;
        
    }
}

/* ----- OR -------------- */
/* O(NlogN) */

class Solution {
    public int[] sortedSquares(int[] nums) {
        for(int i=0; i< nums.length; i++)
        {
            nums[i] *= nums[i];
        }
        Arrays.sort(nums);
        return nums;
        
    }
}